src/pages/api/auth/[...nextauth].ts

```diff
@@ -1,15 +1,34 @@
+import createFetch from '@vercel/fetch'
 import NextAuth from 'next-auth'
 import CloudIdpProvider from 'lib/auth/cloud-idp-provider'
 import refreshTokenSet from 'lib/auth/refresh-token-set'
 import isJwtExpired from 'lib/auth/is-jwt-expired'
 import { AuthErrors } from 'types/auth'
 
+const fetch = createFetch()
+
 const isDev = process.env.NODE_ENV === 'development'
 
 export default NextAuth({
 	session: {
 		maxAge: 2592000, // 30 days
 	},
+	// https://next-auth.js.org/configuration/options#events
+	events: {
+		/**
+		 * NOTE: NextAuth log out of auth providers, so we have to handle doing that
+		 * ourselves in this signOut event.
+		 * https://github.com/nextauthjs/next-auth/discussions/3938
+		 *
+		 * cloud-idp and Ory Hydra docs:
+		 * https://hashicorp.atlassian.net/wiki/spaces/CLOUD/pages/2321449597/cloud-idp+and+Ory+Hydra
+		 */
+		async signOut({ token }) {
+			await fetch(
+				`${__config.dev_dot.auth.idp_url}/oauth2/sessions/logout?id_token_hint=${token.id_token}`
+			)
+		},
```

Comment:
nit: Should we wrap this in a `try/catch`? Or is there not really a meaningful way to handle the error here?

If we do not end up doing anything with the result here, we can just return the fetch call:
```suggestion
		signOut({ token }) {
			return fetch(
				`${__config.dev_dot.auth.idp_url}/oauth2/sessions/logout?id_token_hint=${token.id_token}`
			)
		},
```

Reply:
I tried returning the `await fetch` originally, but there's a TS issue with that:

```
Type '({ token }: { session: Session; token: JWT; }) => Promise<Response>' is not assignable to type '(message: { session: Session; token: JWT; }) => Awaitable<void>'.
  Type 'Promise<Response>' is not assignable to type 'Awaitable<void>'.
    Type 'Promise<Response>' is not assignable to type 'PromiseLike<void>'.
      Types of property 'then' are incompatible.
        Type '<TResult1 = Response, TResult2 = never>(onfulfilled?: (value: Response) => TResult1 | PromiseLike<TResult1>, onrejected?: (reason: any) => TResult2 | PromiseLike<...>) => Promise<...>' is not assignable to type '<TResult1 = void, TResult2 = never>(onfulfilled?: (value: void) => TResult1 | PromiseLike<TResult1>, onrejected?: (reason: any) => TResult2 | PromiseLike<TResult2>) => PromiseLike<...>'.
          Types of parameters 'onfulfilled' and 'onfulfilled' are incompatible.
            Types of parameters 'value' and 'value' are incompatible.
              Type 'Response' is not assignable to type 'void'.ts(2322)
```

Reply:
Wrapped in a `try/catch` here! I think we'll be able to catch these errors in datadog now, right? https://github.com/hashicorp/dev-portal/pull/1301/commits/d85fd5623db7ff2137f3e2fff7b2e3983eb24647

Reply:
@ashleemboyer yeah, datadog will get these logs!